app/src/test/java/org/mozilla/fenix/compose/cfr/CFRPopupFullScreenLayoutTest.kt

```diff
@@ -0,0 +1,100 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.cfr
+
+import android.content.Context
+import android.graphics.PixelFormat
+import android.view.View
+import android.view.ViewManager
+import android.view.WindowManager
+import android.view.WindowManager.LayoutParams
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import androidx.savedstate.ViewTreeSavedStateRegistryOwner
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.spyk
+import io.mockk.verify
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
+
+@RunWith(FenixRobolectricTestRunner::class)
+class CFRPopupFullScreenLayoutTest {
+    @Test
+    fun `WHEN the popup is constructed THEN setup lifecycle owners`() {
+        val container = View(testContext)
+
+        val popupView = spyk(CFRPopupFullScreenLayout("", container, mockk(), mockk()) {})
+
+        assertEquals(
+            ViewTreeLifecycleOwner.get(container),
+            ViewTreeLifecycleOwner.get(popupView)
+        )
+        assertEquals(
+            ViewTreeSavedStateRegistryOwner.get(container),
+            ViewTreeSavedStateRegistryOwner.get(popupView)
+        )
+    }
+
+    @Test
+    fun `WHEN the popup is dismissed THEN cleanup lifecycle owners and detach from window`() {
+        val context = spyk(testContext)
+        val anchor = View(context)
+        val windowManager = spyk(context.getSystemService(Context.WINDOW_SERVICE) as WindowManager)
+        every { context.getSystemService(Context.WINDOW_SERVICE) } returns windowManager
+        val popupView = CFRPopupFullScreenLayout("", anchor, mockk(), mockk()) {}
+        popupView.show()
+
+        popupView.dismiss()
+
+        assertEquals(null, ViewTreeLifecycleOwner.get(popupView))
```

Comment:
This check doesn't seem to fulfill its purpose since it is null before `.dismiss()` and `.show()` as well

Reply:
If we want to check if `ViewTreeLifecycleOwner.set(this, null)` was called `onDismiss()`, wouldn't mocking static methods and checking if it was called with `null` lifecycle do the trick?

Reply:
Thank you!
Same as above we can easily test even with a real initial owner that is actually set to null after the call to `dismiss`.